core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/PathTrackingInputFormat.java

```diff
@@ -129,8 +152,83 @@ public float getProgress() throws IOException, InterruptedException {
     }
 
     @Override
+    public boolean nextKeyValue() throws IOException, InterruptedException {
+      return delegate.nextKeyValue();
+    }
+
+    @Override
     public void close() throws IOException {
       delegate.close();
     }
   }
+
+  private static class TrackingTextRecordReader extends TrackingRecordReader<LongWritable, Text> {
+
+    private TrackingTextRecordReader(RecordReader<LongWritable, Text> delegate, @Nullable String pathField,
+                                     String path) {
+      super(delegate, pathField, path, getTextOutputSchema(pathField));
```

Comment:
you should pass in the schema set in the config, not generate it here. Otherwise the schema set by the inputformat is different than the schema used here, which makes it easy to introduce bugs later on.